### PR TITLE
[FLINK-16668][python] Improve PythonDriver to parse python dependency info from configuration

### PR DESCRIPTION
**Pull request by WeiZhong94**
## What is the purpose of the change

*This pull request improves PythonDriver to parse python dependency info from configuration.*

## Brief change log

  - *Let `PythonDriver` read python dependency info from configuration.*
  - *Refactor `PythonDriverEnvUtils#prepareEnvironment` to construct `PythonEnvironment` object from configuration.*

## Verifying this change

This change is already covered by existing tests, such as *PythonDriverTest*, *PythonDriverEnvUtilsTest*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by WeiZhong94** [comment]
@dianfu Thanks for your review! I have updated this PR according to your comments.

File: flink-python/src/main/java/org/apache/flink/client/python/PythonDriver.java [hunk 1]
```diff
@@ -59,6 +61,8 @@ public static void main(String[] args) {
 			System.exit(1);
 		}
 
+		Configuration config = ExecutionEnvironment.getExecutionEnvironment().getConfiguration();
```
Comment (dianfu) [inline flink-python/src/main/java/org/apache/flink/client/python/PythonDriver.java:64]: Does this work for streaming job?

File: flink-python/src/test/java/org/apache/flink/client/python/PythonDriverEnvUtilsTest.java [hunk 2]
```diff
@@ -34,108 +36,88 @@
...
 
 	private String tmpDirPath;
+	private static String workingDir = new File("").getAbsolutePath();
 
 	@Before
 	public void prepareTestEnvironment() {
 		File tmpDirFile = new File(System.getProperty("java.io.tmpdir"), "pyflink_" + UUID.randomUUID());
 		tmpDirFile.mkdirs();
 		this.tmpDirPath = tmpDirFile.getAbsolutePath();
+		if (new File("flink-python/lib").exists()) {
```
Comment (dianfu) [inline flink-python/src/test/java/org/apache/flink/client/python/PythonDriverEnvUtilsTest.java:67]: This is a little hack for me. Could we avoid these kind of hacks?

File: flink-python/src/test/java/org/apache/flink/client/python/PythonDriverEnvUtilsTest.java [hunk 3]
```diff
@@ -34,108 +36,88 @@
...
+import static org.apache.flink.python.util.PythonDependencyUtils.FILE_DELIMITER;
+
 /**
  * Tests for the {@link PythonDriverEnvUtils}.
  */
 public class PythonDriverEnvUtilsTest {
-	private static final String UUID_PATTERN = "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}";
 
 	private String tmpDirPath;
+	private static String workingDir = new File("").getAbsolutePath();
```
Comment (dianfu) [inline flink-python/src/test/java/org/apache/flink/client/python/PythonDriverEnvUtilsTest.java:60]: the `workingDir` could be a local variable

File: flink-python/src/main/java/org/apache/flink/client/python/PythonDriver.java (hunk 1 above)
Comment (WeiZhong94) [reply flink-python/src/main/java/org/apache/flink/client/python/PythonDriver.java:64]: Yes, the configurations of streaming and batch environments are always set at the same time.

**Comment by WeiZhong94** [comment]
@dianfu Thanks for your review. I have updated this PR according to your comments.

**Comment by dianfu** [comment]
Thanks a lot for the update. LGTM. Merging...

**Pull request closed without merging**